util/network-devp2p/src/discovery.rs

```diff
@@ -239,8 +250,12 @@ impl<'a> Discovery<'a> {
 	}
 
 	fn discover(&mut self) {
-		self.update_new_nodes();
-		if self.discovery_round == DISCOVERY_MAX_STEPS {
+		if self.discovery_round.is_none() {
+			return;
+		}
+		let discovery_round = self.discovery_round.expect("discovery_round is not None ; qed");
```

Comment:
Could use an `if let` pattern, but at the same time it indents +1 , keeping it as it is seems fine.

Reply:
Can be this though: 
```rust
let discovery_round = match self.discovery_round {
	Some(r) => r,
	None => return,
};
```